sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs

```diff
@@ -229,36 +232,158 @@ protected virtual ValueTask<bool> IsEnvironmentReadyAsync()
-            TimeSpan delay = TimeSpan.FromSeconds(10);
-            for (int i = 0; i < numberOfTries; i++)
+            if (GlobalIsRunningInCI)
             {
-                var isReady = await IsEnvironmentReadyAsync();
-                if (isReady)
+                if (Mode == RecordedTestMode.Live)
                 {
-                    return;
+                    int numberOfTries = 60;
+                    TimeSpan delay = TimeSpan.FromSeconds(10);
+                    for (int i = 0; i < numberOfTries; i++)
+                    {
+                        var isReady = await IsEnvironmentReadyAsync();
+                        if (isReady)
+                        {
+                            return;
+                        }
+
+                        await Task.Delay(delay);
+                    }
+
+                    throw new InvalidOperationException(
+                        "The environment has not become ready, check your TestEnvironment.IsEnvironmentReady scenario.");
                 }
-                await Task.Delay(delay);
             }
+            else
+            {
+                await ExtendResourceGroupExpirationAsync();
+            }
+        }
+
+        private async Task ExtendResourceGroupExpirationAsync()
+        {
+            if (Mode is not RecordedTestMode.Live or RecordedTestMode.Record)
+            {
+                return;
+            }
+
+            // determine whether it is even possible to talk to the management endpoint
+            string resourceGroup = GetOptionalVariable("RESOURCE_GROUP");
+            if (resourceGroup == null)
+            {
+                return;
+            }
+
+            string subscription = GetOptionalVariable("SUBSCRIPTION_ID");
+            if (subscription == null)
+            {
+                return;
+            }
+
+            string tenantId = GetOptionalVariable("TENANT_ID");
+            string clientId = GetOptionalVariable("CLIENT_ID");
+            string clientSecret = GetOptionalVariable("CLIENT_SECRET");
+            string authorityHost = GetOptionalVariable("AZURE_AUTHORITY_HOST");
+
+            if (tenantId == null || clientId == null || clientSecret == null || authorityHost == null)
+            {
+                return;
+            }
+
+            // intentionally not using the Credential property as we don't want to throw if the env vars are not available, and we want to allow this to vary per environment.
+            var credential = new ClientSecretCredential(
+                    tenantId,
+                    clientId,
+                    clientSecret,
+                    new ClientSecretCredentialOptions()
+                    {
+                        AuthorityHost = new Uri(authorityHost)
+                    });
+            HttpPipeline pipeline = HttpPipelineBuilder.Build(ClientOptions.Default, new BearerTokenAuthenticationPolicy(credential, "https://management.azure.com/.default"));
```

Comment:
Would there be any value in creating a mini-client to do this, rather than writing out the pipeline code directly?  This is almost a LLC ...

Reply:
Maybe when we have an additional use case 😄